File: authentication-server/src/main/java/com/andrew/authenticationserver/controller/RegistrationController.java
package com.andrew.authenticationserver.controller;

import com.andrew.authenticationserver.entity.dtos.PasswordDto;
import com.andrew.authenticationserver.entity.dtos.RegistrationUserDto;
import com.andrew.authenticationserver.service.registration.RegistrationService;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@AllArgsConstructor
@RequestMapping("/registration")
public class RegistrationController {
    private final RegistrationService registrationService;


    @PostMapping("/send-ver-code")
    public ResponseEntity<?> send(@RequestBody PasswordDto passwordDto,
                                  @RequestParam(name="email") String email){
        return registrationService.sendEmailCode(passwordDto, email);
    }

    @PostMapping("/add-user")
    public ResponseEntity<?> info(@RequestBody RegistrationUserDto registrationUserDto){
        return registrationService.registrationUser(registrationUserDto);
    }
}

File: auth-server/src/main/java/com/andrew/authserver/controller/RegistrationController.kt
package com.andrew.authserver.controller

import com.andrew.authserver.entity.dtos.RegistrationUserDto
import com.andrew.authserver.service.reg_serv.RegistrationService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/registration")
open class RegistrationController(
        private var registrationService: RegistrationService
) {
    @PostMapping("/send-code")
    fun senMail(@RequestBody userDto: RegistrationUserDto){
        registrationService.sendEmailCode(userDto);
    }
}